src/cachingreader.h

```diff
@@ -34,6 +35,8 @@ typedef struct Hint {
     int priority;
 } Hint;
 
+typedef QVarLengthArray<Hint, 512> HintVector;
```

Comment:
BTW the individual commits in this PR have justifications for each change: https://github.com/rryan/mixxx/commit/b249ef83a1d692d435f5ec877fd4318d752bfe45

We were spending about 1% of samples in callback (as measured by my profiler) touching the QVector where I would have expected a fraction of a percent. The problem is that QVector::clear frees the underlying memory so each time we had to malloc a new buffer for it and sometimes grow it. I think the documentation around QVector::clear and QVector::reserve is really unclear so it would be nice to avoid it when QVarLengthArray is a little more deterministic (I've read the source and it's pretty simple -- basically it has a member variable array that is the size of whatever your prealloc value is. if you overflow that then it allocates a new buffer and copies everything just as QVector would have). I'm not using it for its suggested benefit of usage on the stack -- I'm mostly using it since its clear method does not free the internal buffer. 

I think that ```QVector::resize(0)``` is the right way to clear the element without freeing the backing store but I think the difference between clear and resize is a landmine of sorts for future developers. 

An additional gotcha to QVector is that it uses implicit data sharing. If the reference count of the vector rises above 1 then any non-const method called on the vector will clone it (that's Qt's copy-on-write mechanism at work). We had instances of this in the callback as well where using ```operator[]``` instead of ```at``` was causing a clone of the list or vector to be done every frame. QVarLengthArray does not use implicit data sharing so we don't have to worry about it for that.